umpost/um2netcdf.py

```diff
@@ -603,22 +594,58 @@ def check_pressure_warnings(need_heaviside_uv, heaviside_uv, need_heaviside_t, h
               "These fields will be skipped")
 
 
+def filtered_cubes(cubes, include=None, exclude=None):
+    """
+    Generator filters & emits cubes by include or exclude lists.
+
+    Include & exclude args are mutually exclusive. If neither include or exclude
+    are specified, the generator yields the full cube list.
+
+    Parameters
+    ----------
+    cubes : Sequence of Iris Cube objects
+    include: Sequence of item_code (int) to include (discarding all others)
+    exclude: Sequence of item_code (int) to exclude (keeping all other cubes)
+    """
+    if include and exclude:
+        msg = "Include and exclude lists are mutually exclusive"
+        raise ValueError(msg)
+
+    f_cubes = None
+
+    if include is None and exclude is None:
+        f_cubes = cubes
+    elif include:
+        f_cubes = (c for c in cubes if c.item_code in include)
+    elif exclude:
+        f_cubes = (c for c in cubes if c.item_code not in exclude)
+
+    for c in f_cubes:
+        yield c
+
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description="Convert UM fieldsfile to netcdf")
     parser.add_argument('-k', dest='nckind', required=False, type=int,
                         default=3,
-                        help='specify netCDF output format: 1 classic, 2 64-bit offset, 3 netCDF-4, 4 netCDF-4 classic model. Default 3',
+                        help=('specify netCDF output format: 1 classic, 2 64-bit'
+                              ' offset, 3 netCDF-4, 4 netCDF-4 classic model.'
+                              ' Default 3'),
                         choices=[1, 2, 3, 4])
     parser.add_argument('-c', dest='compression', required=False, type=int,
                         default=4, help='compression level (0=none, 9=max). Default 4')
     parser.add_argument('--64', dest='use64bit', action='store_true',
                         default=False, help='Use 64 bit netcdf for 64 bit input')
     parser.add_argument('-v', '--verbose', dest='verbose',
-                        action='count', default=0, help='verbose output (-vv for extra verbose)')
-    parser.add_argument('--include', dest='include_list', type=int,
-                        nargs='+', help='List of stash codes to include')
-    parser.add_argument('--exclude', dest='exclude_list', type=int,
-                        nargs='+', help='List of stash codes to exclude')
+                        action='count', default=0,
+                        help='verbose output (-vv for extra verbose)')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--include', dest='include_list', type=int,
+                       nargs='+', help='List of stash codes to include')
+    group.add_argument('--exclude', dest='exclude_list', type=int,
+                       nargs='+', help='List of stash codes to exclude')
+
```

Comment:
Given the growing size of the file and the numerous arguments, it might be beneficial to move the CLI arguments to a separate file in the future. This approach aligns with the separation of concerns principle, making the file easier to maintain. Additionally, this would be particularly useful if the library is also invoked by other Python programs, not just from the CLI.

Reply:
I've added that idea in #27.